src/Features/Core/Portable/RemoveUnusedParametersAndValues/AbstractRemoveUnusedParametersAndValuesDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,856 @@
+                        value.ConstantValue.HasValue ||
+                        value is IAssignmentOperation ||
+                        value is IIncrementOrDecrementOperation ||
+                        value is IInvalidOperation ||
+                        value.Syntax.GetFirstToken() != expressionStatement.Syntax.GetFirstToken() ||
+                        (value is IInvocationOperation invocation &&
+                         (invocation.TargetMethod.ContainingType.OriginalDefinition == _symbolStartAnalyzer._interlockedType ||
+                          invocation.TargetMethod.ContainingType.OriginalDefinition == _symbolStartAnalyzer._immutableInterlockedType)))
+                    {
+                        return;
+                    }
+
+                    // "Expression value is never used"
+                    var properties = s_propertiesMap[(_options.UnusedValueExpressionStatementPreference, isUnusedLocalAssignment: false, isRemovableAssignment: false)];
+                    var diagnostic = DiagnosticHelper.Create(s_expressionValueIsUnusedRule,
+                                                             value.Syntax.GetLocation(),
+                                                             _options.UnusedValueExpressionStatementSeverity,
+                                                             additionalLocations: null,
+                                                             properties);
+                    context.ReportDiagnostic(diagnostic);
+                }
+
+                private void AnalyzeDelegateCreationOrAnonymousFunction(OperationAnalysisContext operationAnalysisContext)
+                    => _hasDelegateCreationOrAnonymousFunction = true;
+
+                private void AnalyzeConversion(OperationAnalysisContext operationAnalysisContext)
+                {
+                    if (_hasConversionFromDelegateTypeToNonDelegateType)
+                    {
+                        return;
+                    }
+
+                    var conversion = (IConversionOperation)operationAnalysisContext.Operation;
+                    if (conversion.Operand.Type.IsDelegateType() &&
+                        !conversion.Type.IsDelegateType())
+                    {
+                        _hasConversionFromDelegateTypeToNonDelegateType = true;
+                    }
+                }
+
+                private void AnalyzeFieldOrPropertyReference(OperationAnalysisContext operationAnalysisContextContext)
+                {
+                    var fieldOrPropertyReference = operationAnalysisContextContext.Operation;
+                    if (fieldOrPropertyReference.Type.IsDelegateType() &&
+                        fieldOrPropertyReference.Parent is ISimpleAssignmentOperation simpleAssignment &&
+                        simpleAssignment.Target == fieldOrPropertyReference)
+                    {
+                        _escapedDelegate = true;
```

Comment:
1. doesn't follow pattern of method right above this.  if this is also true, no need to analyze again.